datadog_checks_base/datadog_checks/utils/decorator.py

```diff
@@ -0,0 +1,22 @@
+# (C) Datadog, Inc. 2018
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+from datadog_checks.config import is_affirmative
+from ddtrace import tracer
+from functools import wraps
+from inspect import getargspec
+
+def trace_func(func):
+    @wraps(func)
+    def function_wrapper(*args, **kwargs):
+        # Get instance config to see if tracing is enabled
+        try:
+            instance_index = getargspec(func).args.index('instance')
+        except ValueError:
+            return func(*args, **kwargs)
+        if is_affirmative(args[instance_index].get('trace_check', False)):
+            with tracer.trace('integration.check', service='IntegrationsTracing', resource=args[0].name):
```

Comment:
`args[0].name` is the name of the check, right? About the service name, you can use `integrations-tracing` just because usually we use that naming convention.

Reply:
Yeah, args[0] is just the name of the check here, in this case it comes in the UI as `openstack`. I updated the service name to match that convention ^